lambdas/send_from_queue/send_from_queue.py

```diff
@@ -208,19 +208,34 @@ def lambda_handler(event, _context):
                     "queue. {}".format(message_body)
                 )
 
+            api_url_full_query = api_url
+            already_triggered = message_body.get("already_triggered")
+            if already_triggered:
+                    api_url_full_query = api_url_full_query + \
+                        "&already_triggered={}".format(already_triggered)
+
             # Send the SQS message to the API Gateway
             response = post(
-                api_url_query,
+                api_url_full_query,
                 json=payload,
                 headers=headers,
                 timeout=(3.05, 30)
             )
 
-            # If there was a problem, raise an error
-            response.raise_for_status()
-
-            # Otherwise, delete the message since it has been processed
-            _delete_from_queue(queue_object, message)
+            error_message = "More jobs triggered, but " \
+                            "unable to trigger all jobs."
+            if error_message in response.text:
+                # The hook failed to trigger all jobs, but it has
+                # been replaced in the queue with a new hook, so
+                # delete this one.
+                _delete_from_queue(queue_object, message)
+                response.raise_for_status()
+            else:
+                # If there was a problem, raise an error
+                response.raise_for_status()
+
+                # Otherwise, delete the message since it has been processed
+                _delete_from_queue(queue_object, message)
```

Comment:
I am confused by the logic here: it looks like we are doing the same thing in both if blocks. Also, exceptions are potentially thrown here that might not make sense. What do you think about wrapping it in a try/catch instead, and bubble exceptions up with a more meaningful message to a log reader